src/replay/trackers.zig

```diff
@@ -17,27 +17,90 @@ const SlotState = sig.core.SlotState;
 /// kitchen-sink style approach of storing everything under the sun.
 ///
 /// [BankForks](https://github.com/anza-xyz/agave/blob/161fc1965bdb4190aa2d7e36c7c745b4661b10ed/runtime/src/bank_forks.rs#L75)
+///
+/// This struct is *not* thread safe, and the lifetimes of the returned pointers
+/// will end as soon as the items are removed.
 pub const SlotTracker = struct {
-    slots: std.AutoArrayHashMapUnmanaged(Slot, Element) = .{},
+    slots: std.AutoArrayHashMapUnmanaged(Slot, *Element),
+    root: std.atomic.Value(Slot),
```

Comment:
Rather not make this atomic; if SlotTracker is not already properly internally locking, and not exposing a well defined API to access it safely accross threads, I don't think it's a sound decision to create a false sense of security or lattice of "some parts are thread safe, others are not".